### ISSUE TITLE
error  'use strict' is unnecessary inside of modules

### ISSUE THREAD

**Issue by mastertinner** [eslint/eslint#5301]
I am writing Node.js code and am getting `error  'use strict' is unnecessary inside of modules` because i use `'use strict';` at the top of all my files. However, if I remove the `'use strict';`, then it tells me that I cannot use `let` or `const` because they are only available in strict mode. Can you tell me how to solve this?

I am using https://github.com/mastertinner/eslint-config-mastertinner-nodejs for the configuration and eslint `2.1.0`.

**Comment by albertosantini** [comment]
Related with https://github.com/eslint/eslint/issues/4936

**Comment by IanVS** [comment]
@mastertinner You have `sourceType: "module"` set in your config, therefore you get `strict` mode for free (meaning, you shouldn't use the `"use strict"` directive).  What is the warning you are getting about `let` and `const`?

**Comment by eslintbot** [comment]
Hi @mastertinner, thanks for the issue. It looks like there's not enough information for us to know how to help you. If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by mastertinner** [comment]
@IanVS, the error I'm getting is the following
```
let redisUrl = '';
^^^

SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode
```

and it's about the following file:
```javascript
const cfEnv = require('cfenv');
const appEnv = cfEnv.getAppEnv();

let redisUrl = '';

if (process.env.VCAP_SERVICES) {
  const credentials = appEnv.getService('radiator-sync').credentials;

  redisUrl = `redis://:${credentials.password}@${credentials.host}:${credentials.port}`;
} else {
  redisUrl = 'redis://localhost:6379';
}

module.exports = redisUrl;
```

The same happens when I try to use `class`. Oddly enough, there are files where I use `const` but there I'm not getting an error (even without `'use strict';`).

**Comment by nre** [comment]
@mastertinner, `sourceType: "module"` is only for ES6 modules (`export`, etc). Your modules are Node.js modules (`module.exports`, etc). You should keep the strict-mode directives in your modules, and remove `sourceType: "module"` from your config.

(I made a similar mistake in #4832).

**Comment by mastertinner** [comment]
Thanks so much, @nre! That solved it.
